crates/sui-json-rpc/src/name_service.rs

```diff
@@ -60,6 +64,33 @@ impl Domain {
             type_params: vec![],
         }
     }
+
+    /// Returns the SLD (second level domain) for a domain.
+    /// E.g. `test.test.test.example.sui` -> `example.sui`
+    pub fn sld(&self) -> Domain {
+        Domain {
+            labels: self.labels[0..2].to_vec(),
+        }
+    }
+
+    /// Derive the parent domain for a given domain
+    /// E.g. `test.example.sui` -> `example.sui`
+    pub fn parent(&self) -> Domain {
+        Domain {
+            labels: self.labels[0..(self.labels.len() - 1)].to_vec(),
+        }
+    }
```

Comment:
For these indexing operations, can you document what safety property makes sure that these won't panic? e.g. how do we know that a domain always has at least `2` elements?

Reply:
This is invoked only for subdomains (labels length >=3). 
Do you think it's generally better to change to a Result(), and check the length to be safer?

Reply:
I actually ended up adding a check when constructing domains, to make sure that the length is at least 2 (Since we won't ever have a TLD pointing somewhere anyways).